src/puck.ts

```diff
@@ -29,6 +29,22 @@ export interface PuckMouseEvent extends MouseEvent {
   fromPuck: true;
 }
 
+type ClickState =
+  | {
+      kind: 'idle';
+    }
+  | {
+      kind: 'gotpointerdown';
+    }
+  | {
+      kind: 'firstclick';
+      timeout: number;
+    }
+  | {
+      kind: 'firstclickwithpointerdown';
+      timeout: number;
+    };
```

Comment:
As mentioned in https://github.com/birchill/10ten-ja-reader/pull/740#issuecomment-908260962 I think we might want to cover dragging here so that we end up with something like:

```typescript
type ClickState =
  | {
      kind: 'idle';
    }
  | {
      kind: 'firstpointerdown';
      // This is the timeout we use to detect if it's a drag or not
      timeout: number;
    }
  | {
      kind: 'dragging';
    }
  | {
      kind: 'firstclick';
      // This is the timeout we use to detect if it's a double-click or not
      timeout: number;
    }
  | {
      kind: 'secondpointerdown';
      // This is the same timeout as we start when we enter the firstclick state
      timeout: number;
    };
```

Reply:
Implemented in 779a08a.